state-sync/src/coordinator.rs

```diff
@@ -1103,22 +1097,36 @@ impl<T: ExecutorProxyTrait> StateSyncCoordinator<T> {
             self.local_state.trusted_epoch()
         };
         self.local_state.verify_ledger_info(&response_li)?;
-        if let Some(li) = pending_li {
-            if li != response_li {
-                self.local_state.verify_ledger_info(&li)?;
+        self.validate_and_store_chunk(txn_list_with_proof, response_li.clone(), None)?;
+
+        // Need to sync with local storage to update synced version
+        self.sync_state_with_local_storage()?;
+        let synced_version = self.local_state.synced_version();
+
+        // Check if we've synced beyond our current highest known ledger info
+        if let Some(highest_known_ledger_info) = &self.highest_known_ledger_info {
+            if synced_version >= highest_known_ledger_info.ledger_info().version() {
+                self.highest_known_ledger_info = None;
             }
-            self.pending_ledger_infos.add_li(li);
         }
-        self.validate_and_store_chunk(txn_list_with_proof, response_li, None)?;
 
-        // need to sync with local storage to see whether response LI was actually committed
-        // and update pending_ledger_infos accordingly
-        self.sync_state_with_local_storage()?;
-        let new_version = self.local_state.synced_version();
+        // If we don't have a highest known ledger info, check if the new highest
+        // is appropriate for us.
+        if self.highest_known_ledger_info.is_none() {
```

Comment:
Can you elaborate on what you mean by this? Merging the two checks for self.highest_known_ledger_info? If so, we don't want to use an `if-else` structure, because we need both branches to be executed.